src/steps/object-delete.ts

```diff
@@ -0,0 +1,45 @@
+import { BaseStep, Field, StepInterface, ExpectedRecord } from '../core/base-step';
+import { Step, RunStepResponse, FieldDefinition, StepDefinition, RecordDefinition } from '../proto/cog_pb';
+
+export class DeleteObject extends BaseStep implements StepInterface {
+
+  protected stepName: string = 'Delete a Salesforce Object';
+  protected stepExpression: string = 'delete the salesforce (?<objName>[a-zA-Z0-9]+) object with id (?<id>.+)';
+  protected stepType: StepDefinition.Type = StepDefinition.Type.ACTION;
+  protected expectedFields: Field[] = [{
+    field: 'objName',
+    type: FieldDefinition.Type.STRING,
+    description: 'Salesforce Object name',
+  }, {
+    field: 'id',
+    type: FieldDefinition.Type.STRING,
+    description: 'Object ID',
+  }];
+  protected expectedRecords: ExpectedRecord[] = [{
+    id: 'salesforceObject ',
+    type: RecordDefinition.Type.KEYVALUE,
+    fields: [{
+      field: 'Id',
+      type: FieldDefinition.Type.STRING,
+      description: "Object's SalesForce ID",
+    }],
+    dynamicFields: false,
+  }];
+
+  async executeStep(step: Step): Promise<RunStepResponse> {
+    const stepData: any = step.getData().toJavaScript();
+    const objName: any = stepData.objName;
+    const id: any = stepData.id;
+
+    try {
+      const result = await this.client.deleteObjectById(objName, id);
+      const record = this.keyValue('object', 'Deleted Object', { Id: result.id });
+      return this.pass('Successfully deleted Object %s', [result.id], [record]);
```

Comment:
- [ ] Same as above: consider adding the `objName` to make the message friendlier.